packages/react-topology/src/components/nodes/DefaultNode.tsx

```diff
@@ -192,9 +204,52 @@ const DefaultNode: React.FunctionComponent<DefaultNodeProps> = ({
   }
 
   const nodeLabelPosition = labelPosition || element.getLabelPosition();
+  const scale = element.getGraph().getScale();
+
+  React.useEffect(() => {
+    let handle: NodeJS.Timeout;
+
+    if (!scaleNode || scale >= 1) {
+      setNodeScale(1);
+    } else {
+      const scaleGoal = 1 / scale;
+      const scaleBump = (scaleGoal - 1) / SCALE_UP_STEPS;
+      const bumpScale = (scaleStep: number = 0) => {
+        setNodeScale(1 + scaleBump * scaleStep);
+        if (scaleStep < SCALE_UP_STEPS) {
+          handle = setTimeout(() => {
+            bumpScale(scaleStep + 1);
+          }, SCALE_UP_INTERVAL / SCALE_UP_STEPS);
+        }
+      };
```

Comment:
I find the scaling animation a bit slow.
You should use `requestAnimationFrame` instead of timeout for animations.

Furthermore the scaling value should be based on time and not step intervals.
That way the scaling timing is consistent always and let `requestAnimationFrame` dictate how many frames we can actually render smoothly in the allotted time.

Reply:
Updated and handled unmount.